src/metabase/events/schema.clj

```diff
@@ -10,10 +10,14 @@
                       [:map {:closed true}
                        [:user-id pos-int?]
                        [:object [:fn #(t2/instance-of? :model/Dashboard %)]]])
+      view-only      (mc/schema
+                      [:map {:closed true}
```

Comment:
nit: it was a little unclear to me what `view-only` means—the only difference is that `user-id` is optional, right?

I think it would actually be fine to just change the default-schema to take an optional user ID, but up to you